packages/realm-web/src/App.ts

```diff
@@ -111,6 +111,16 @@ export class App<
    */
   public readonly authenticator: Authenticator;
 
+  /**
+   * an extended property that may be set/removed on this object statically,
+   *  or defined as a dynamic getter function
+   * will be sent with mongoClient and functions requests
+   *
+   * @type {Headers}
+   * @memberof User
+   */
+  public headers?: Headers;
```

Comment:
I would like this to be a `private` member of `App` only available though the configuration passed to the constructor.

Reply:
To be accessible from Fetcher, the property needs to be public in the LocationUrlContext and UserContext interfaces. I propose "public readonly" prop to be set by constructor option in Fetcher and App classes (also `clone` option for Fetcher).  User class will have a public getter, no setter, and utility methods for adding or removing key/value pairs.